internal/server/redis/redis_test.go

```diff
@@ -0,0 +1,113 @@
+package redis
+
+import (
+	"context"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/alicebob/miniredis/v2"
```

Comment:
The main advantage is speed and not needing to start a redis instance in order to run tests. This makes writing tests easy and fast. 

I haven't used `miniredis` previous but @pdevine has and he seems happy with it. I've also used Python's `redislite` previously for unit tests and it's a similar idea. I'm not that concerned with compatibility since the Redis API itself is relatively simple and we're not using any advanced features (yet). We can move on to an external Redis if it ever becomes a problem.